src/bcrypt_node.cc

```diff
@@ -63,86 +60,72 @@ bool ValidateSalt(const char* salt) {
 
 /* SALT GENERATION */
 
-class SaltAsyncWorker : public Nan::AsyncWorker {
-public:
-    SaltAsyncWorker(Nan::Callback *callback, std::string seed, ssize_t rounds)
-        : Nan::AsyncWorker(callback), seed(seed), rounds(rounds) {
-    }
-
-    ~SaltAsyncWorker() {}
+class SaltAsyncWorker : public Napi::AsyncWorker {
+    public:
+        SaltAsyncWorker(Napi::Function& callback, std::string seed, ssize_t rounds)
+            : Napi::AsyncWorker(callback), seed(seed), rounds(rounds) {
+        }
 
-    void Execute() {
-        char salt[_SALT_LEN];
-        bcrypt_gensalt(rounds, (u_int8_t *)&seed[0], salt);
-        this->salt = std::string(salt);
-    }
+        ~SaltAsyncWorker() {}
 
-    void HandleOKCallback() {
-        Nan::HandleScope scope;
+        void Execute() {
+            char salt[_SALT_LEN];
+            bcrypt_gensalt((char) rounds, (u_int8_t *)&seed[0], salt);
+            this->salt = std::string(salt);
+        }
 
-        Local<Value> argv[2];
-        argv[0] = Nan::Undefined();
-        argv[1] = Nan::Encode(salt.c_str(), salt.size(), Nan::BINARY);
-        callback->Call(2, argv);
-    }
+        void OnOK() {
+            Napi::HandleScope scope(Env());   
+            Callback().Call({Env().Undefined(), Napi::String::New(Env(), salt)});
+        }
 
-private:
-    std::string seed;
-    std::string salt;
-    ssize_t rounds;
+    private:
+        std::string seed;
+        std::string salt;
+        ssize_t rounds;
+        
 };
 
-NAN_METHOD(GenerateSalt) {
-    Nan::HandleScope scope;
-
+Napi::Value GenerateSalt(const Napi::CallbackInfo& info) {
     if (info.Length() < 3) {
-        Nan::ThrowTypeError("3 arguments expected");
-        return;
+        throw Napi::TypeError::New(info.Env(), "3 arguments expected");
+        return info.Env().Undefined();
```

Comment:
Why are these returns needed if you're throwing? I see them throughout, but thought they were unnecessary, no?

Reply:
You're right it's not necessary if exceptions system is enable. I provided to remove from the code all unuseful code.